lighthouse-core/gather/fetcher.js

```diff
@@ -172,37 +104,14 @@ class Fetcher {
     };
   }
 
-  /**
-   * @param {string} requestId
-   * @return {Promise<string>}
-   */
-  async _resolveResponseBody(requestId) {
-    const responseBody = await this.session.sendCommand('Fetch.getResponseBody', {requestId});
-    if (responseBody.base64Encoded) {
-      return Buffer.from(responseBody.body, 'base64').toString();
-    }
-    return responseBody.body;
-  }
-
   /**
    * @param {string} url
    * @param {{timeout: number}} options timeout is in ms
    * @return {Promise<FetchResponse>}
    */
   async _fetchResourceOverProtocol(url, options) {
     const startTime = Date.now();
-
-    /** @type {NodeJS.Timeout} */
-    let timeoutHandle;
-    const timeoutPromise = new Promise((_, reject) => {
-      timeoutHandle = setTimeout(reject, options.timeout, new Error('Timed out fetching resource'));
-    });
-
-    const responsePromise = this._loadNetworkResource(url);
-
-    /** @type {{stream: LH.Crdp.IO.StreamHandle|null, status: number|null}} */
-    const response = await Promise.race([responsePromise, timeoutPromise])
-      .finally(() => clearTimeout(timeoutHandle));
+    const response = await this._wrapWithTimeout(this._loadNetworkResource(url), options.timeout);
```

Comment:
`_readIOStream` has its own weird timeout system, would it be possible to remove that and use `_wrapWithTimeout` over `_fetchResourceOverProtocol`?

Reply:
it wouldn't be the same, unless we made the work `_readIOStream` does use a cancel-able promise. even if the wrapped timer timed out the work would keep going